pandas/core/generic.py

```diff
@@ -688,6 +688,92 @@ def f(x):
 
     rename.__doc__ = _shared_docs['rename']
 
+    _shared_docs['relabel'] = """
+        Assign new axes based on list-like labels
+
+        .. versionadded:: 0.20.0
+
+        Parameters
+        ----------
+        %(axes)s : list-like, optional
+            Labels to construct new axis from - number of labels
+            must match the length of the existing axis
+        copy : boolean, default True
+            Also copy underlying data
```

Comment:
Yeah, not sure - I agree it's confusing.  I'd be tempted to deprecate `copy` altogether, though if you know what you're doing I suppose it's useful.